src/main/scala/uk/gov/nationalarchives/downloadfiles/FileUtils.scala

```diff
@@ -0,0 +1,69 @@
+package uk.gov.nationalarchives.downloadfiles
+
+import java.net.URLDecoder
+import java.nio.file.Paths
+import java.util.UUID
+
+import com.amazonaws.services.lambda.runtime.events.models.s3.S3EventNotification.S3EventNotificationRecord
+import com.typesafe.config.{Config, ConfigFactory}
+import graphql.codegen.GetOriginalPath.getOriginalPath.{Data, Variables, document}
+import software.amazon.awssdk.services.s3.S3Client
+import software.amazon.awssdk.services.s3.model.GetObjectRequest
+import sttp.client.{Identity, NothingT, SttpBackend}
+import uk.gov.nationalarchives.tdr.{GraphQLClient, GraphQlResponse}
+import uk.gov.nationalarchives.tdr.error.NotAuthorisedError
+import uk.gov.nationalarchives.tdr.keycloak.KeycloakUtils
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.Try
+
+class FileUtils()(implicit val executionContext: ExecutionContext) {
+  val config: Config = ConfigFactory.load
+
+  def getFilePath(keycloakUtils: KeycloakUtils, client: GraphQLClient[Data, Variables], fileId: UUID)(implicit backend: SttpBackend[Identity, Nothing, NothingT]): Future[Either[String, String]] = {
+
+    val queryResult: Future[Either[String, GraphQlResponse[Data]]] = (for {
+      token <- keycloakUtils.serviceAccountToken(config.getString("auth.client.id"), config.getString("auth.client.secret"))
+      result <- client.getResult(token, document, Option(Variables(fileId)))
+    } yield Right(result)) recover(e => {
+      Left(e.getMessage)
```

Comment:
Here's another exception where the full error is just being ignored.

Reply:
As we discussed before, I've swapped out the Either for Try which will give a `List[Throwable]` once the lambda has processed all of the messages.